File: Examples/ObjcDemo/RixSDKPodDemo/RixEngineObjcDemo/UI/Rixengine/Xib/BannerViewController/AlxBannerXibVC.m
#import "AlxBannerXibVC.h"
#import "AdConfig.h"

@interface AlxBannerXibVC ()

@property (nonatomic, assign) BOOL isLoading;
@property (nonatomic, copy) NSString *TAG;

@end

@implementation AlxBannerXibVC

- (instancetype)init {
    self = [super initWithNibName:@"AlxBannerXibVC" bundle:nil];
    if (self) {
        _TAG = @"Alx-banner:";
        _isLoading = NO;
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    self.navigationItem.title = NSLocalizedString(@"Alx_banner", @"");
    
    [self setupLayout];
}

- (void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
    NSLog(@"%@ viewWillDisappear", self.TAG);
    [self.bannerView destroy];
}

- (void)viewDidDisappear:(BOOL)animated {
    [super viewDidDisappear:animated];
    NSLog(@"%@ viewDidDisappear", self.TAG);
}

- (void)setupLayout {
    [self.bnLoadAndShow setTitle:NSLocalizedString(@"load_and_show_ad", @"") forState:UIControlStateNormal];
    [self.bnLoadAndShow setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    self.bnLoadAndShow.backgroundColor = [UIColor darkGrayColor];
    
    self.label.textAlignment = NSTextAlignmentCenter;
    self.label.textColor = [UIColor blackColor];
    self.label.numberOfLines = 0;
    self.label.text = @"";
}

- (IBAction)bnLoadAndShow:(id)sender {
    if (self.isLoading) {
        return;
    }
    self.isLoading = YES;
    self.label.text = NSLocalizedString(@"loading", @"");
    
    self.bannerView.delegate = self;
    self.bannerView.rootViewController = self;
    self.bannerView.isHideClose = NO;
    [self.bannerView loadAdWithAdUnitId:[AdConfig Alx_Banner_Ad_Id]];
}

#pragma mark - AlxBannerViewAdDelegate

- (void)bannerViewAdLoad:(AlxBannerAdView *)bannerView {
    NSLog(@"%@ bannerViewAdLoad: price: %f", self.TAG, [bannerView getPrice]);
    [bannerView reportBiddingUrl];
    [bannerView reportChargingUrl];
    
    self.isLoading = NO;
    self.label.text = NSLocalizedString(@"load_success", @"");
}

- (void)bannerViewAdFailToLoad:(AlxBannerAdView *)bannerView didFailWithError:(NSError *)error {
    NSString *msg = [NSString stringWithFormat:@"%ld: %@", (long)error.code, error.localizedDescription];
    NSLog(@"%@ bannerViewAdFailToLoad: %@", self.TAG, msg);
    
    self.isLoading = NO;
    self.label.text = [NSString stringWithFormat:NSLocalizedString(@"load_failed", @""), msg];
}

- (void)bannerViewAdImpression:(AlxBannerAdView *)bannerView {
    NSLog(@"%@ bannerViewAdImpression", self.TAG);
}

- (void)bannerViewAdClick:(AlxBannerAdView *)bannerView {
    NSLog(@"%@ bannerViewAdClick", self.TAG);
}

- (void)bannerViewAdClose:(AlxBannerAdView *)bannerView {
    NSLog(@"%@ bannerViewAdClose", self.TAG);
}

@end

File: SwiftPodDemo/AlxAdsDemo/UI/Rixengine/Xib/BannerViewController/AlxBannerXibVC.swift
//
//  AlxBannerXibVC.swift
//  AlxDemo
//
//  Created by liu weile on 2025/5/29.
//
import UIKit
import AlxAds

class AlxBannerXibVC: BaseUIViewController{
    
    private let TAG = "Alx-banner:"
    
    @IBOutlet weak var bnLoadAndShow: UIButton!
    
    @IBOutlet weak var label: UILabel!
    
    
    @IBOutlet weak var bannerView: AlxBannerAdView!
    
    private var isLoading:Bool=false
    
    init() {
        super.init(nibName: "AlxBannerXibVC", bundle: nil)
    }
    
//    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
//        super.init(nibName: "AlxBannerXibVC", bundle: nibBundleOrNil)
//    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
    }
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = .white
        navigationItem.title = NSLocalizedString("Alx_banner", comment: "")
        
        setupLayout()
    }
    
    //视图将要消失【生命周期】
    public override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        print("\(TAG) viewWillDisappear")
        bannerView.destroy()
    }
    
    //视图已经消失【生命周期】
    public override func viewDidDisappear(_ animated: Bool) {
        super.viewDidDisappear(animated)
        print("\(TAG) viewDidDisappear")
    }
    
    private func setupLayout() {
        bnLoadAndShow.setTitle(NSLocalizedString("load_and_show_ad", comment: ""), for: .normal)
        bnLoadAndShow.setTitleColor(.white, for: .normal)
        bnLoadAndShow.backgroundColor = .darkGray
        bnLoadAndShow.accessibilityNavigationStyle = .automatic
        
        label.textAlignment = .center
        label.textColor = .black
        label.numberOfLines = .zero
        label.text = ""
    }
    
    
    @IBAction func bnLoadAndShow(_ sender: Any) {
        if isLoading {
            return
        }
        isLoading=true
        label.text=NSLocalizedString("loading", comment: "")
        
        bannerView.delegate = self
        bannerView.rootViewController = self
        bannerView.isHideClose = false
        bannerView.loadAd(adUnitId: AdConfig.Alx_Banner_Ad_Id)
    }
    
}

extension AlxBannerXibVC:AlxBannerViewAdDelegate {
    
    func bannerViewAdLoad(_ bannerView: AlxBannerAdView) {
        print("\(TAG) bannerViewAdLoad: price: \(bannerView.getPrice())")
        bannerView.reportBiddingUrl()
        bannerView.reportChargingUrl()
        
        self.isLoading=false
        self.label.text=NSLocalizedString("load_success", comment: "")
    }
    
    func bannerViewAdFailToLoad(_ bannerView: AlxBannerAdView, didFailWithError error: Error){
        let error1=error as NSError
        let msg="\(error1.code): \(error1.localizedDescription)"
        print("\(TAG) bannerViewAdFailToLoad: \(msg)")
        
        self.isLoading=false
        self.label.text=String(format: NSLocalizedString("load_failed", comment: ""), msg)
    }
    
    func bannerViewAdImpression(_ bannerView: AlxBannerAdView) {
        print("\(TAG) bannerViewAdImpression")
    }
    
    func bannerViewAdClick(_ bannerView: AlxBannerAdView) {
        print("\(TAG) bannerViewAdClick")
    }
    
    func bannerViewAdClose(_ bannerView: AlxBannerAdView) {
        print("\(TAG) bannerViewAdClose")
    }
    
    
}
